api/core/v1beta2/cluster_types.go

```diff
@@ -1139,16 +1140,19 @@ func (c *ClusterStatus) GetTypedPhase() ClusterPhase {
 // ANCHOR: APIEndpoint
 
 // APIEndpoint represents a reachable Kubernetes API endpoint.
+// +kubebuilder:validation:MinProperties=1
 type APIEndpoint struct {
```

Comment:
@JoelSpeed I think the changes in this PR should also address https://github.com/kubernetes-sigs/cluster-api/issues/6416

And yes we have use cases where folks set either host or port or both

Reply:
I think I agree, provided we are adding `omitzero` at the places where this field is being used.